airbyte-integrations/connectors/source-shopify/source_shopify/streams/base_streams.py

```diff
@@ -716,21 +716,37 @@ def get_updated_state(
         """
         updated_state = super().get_updated_state(current_stream_state, latest_record)
         if self.parent_stream_class:
+            parent_state = latest_record.get(self.parent_stream.name, {})
+            parent_state_value = (
+                parent_state.get(self.parent_stream.cursor_field) if parent_state else latest_record.get(self.parent_stream.cursor_field)
+            )
             # add parent_stream_state to `updated_state`
-            updated_state[self.parent_stream.name] = {self.parent_stream.cursor_field: latest_record.get(self.parent_stream.cursor_field)}
+            updated_state[self.parent_stream.name] = {self.parent_stream.cursor_field: parent_state_value}
         return updated_state
 
-    def get_stream_state_value(self, stream_state: Optional[Mapping[str, Any]]) -> str:
-        if self.parent_stream_class:
-            # get parent stream state from the stream_state object.
-            parent_state = stream_state.get(self.parent_stream.name, {})
-            if parent_state:
-                return parent_state.get(self.parent_stream.cursor_field, self.default_state_comparison_value)
-        else:
-            # get the stream state, if no `parent_stream_class` was assigned.
+    def _get_stream_cursor_value(self, stream_state: Optional[Mapping[str, Any]] = None) -> Optional[str]:
+        if stream_state:
             return stream_state.get(self.cursor_field, self.default_state_comparison_value)
+        else:
+            return self.config.get("start_date")
+
+    def get_stream_state_value(self, stream_state: Optional[Mapping[str, Any]] = None) -> Optional[str]:
+        if stream_state:
+            if self.parent_stream_class:
+                # get parent stream state from the stream_state object.
+                parent_state = stream_state.get(self.parent_stream.name, {})
+                if parent_state:
+                    return parent_state.get(self.parent_stream.cursor_field, self.default_state_comparison_value)
+                else:
+                    # use the streams cursor value, if no parent state available
+                    return self._get_stream_cursor_value(stream_state)
+            else:
+                # get the stream state, if no `parent_stream_class` was assigned.
+                return self._get_stream_cursor_value(stream_state)
+        else:
+            return self.config.get("start_date")
```

Comment:
Why was this case added? I'm unclear as to how this links to the addition of considering the parent in the state

Reply:
Got it! Probably a nit concern but: Is there a case where the stream would have returned an empty state and we could have changed the start_date which would have been applied? The case I see is:
* The user starts a sync with config `start_date = 2024`
* X happens (which I what I'm trying to understand: what would be X?)
* The user changes to `start_date = 2023`
* There is another sync but it start for 2024 since the state fallbacks on the config

Again, nit concern but I still want to raise the question just in case we miss something